src/context/UserContext.js

```diff
@@ -0,0 +1,23 @@
+import { useState } from 'react';
+import { createContext } from 'react';
+import { useContext } from 'react';
+import { getUser } from '../services/user';
+
+const UserContext = createContext();
+
+const UserProvider = ({ children }) => {
+  const [currentUser, setCurrentUser] = useState(getUser() || { email: null });
+  const [userId, setUserId] = useState(getUser());
+
```

Comment:
I believe we can get away with solely using the currentUser instead of having both currentUser and userId.